src/history/historyActions.ts

```diff
@@ -0,0 +1,78 @@
+import { HistoryData } from 'history/types'
+import { Action, ActionCreator } from 'redux'
+import { ThunkAction } from 'redux-thunk'
+import { API, GlobalState } from 'types'
+import { ChangePriceFilterAction } from './historyActions'
+
+/**
+ * TODO
+ * TL 02/10/2018
+ * Could potentially be moved to the root src directory types.ts once all components have been converted to TS?
+ */
+type ThunkResult<R, A extends Action<any>> = ThunkAction<R, GlobalState, API, A>
```

Comment:
A extends Action<any>, by default the interface for Action is 
`Action<T=any>{
 type: T
}`
so you can consider just do A extends Action